scripts/py_matter_yamltests/test_pics_checker.py

```diff
@@ -0,0 +1,164 @@
+\tA.B = 1
+'''
+
+simple_config_with_comments = '''
+# This is a comment
+A.A=0
+# This is an other comment
+A.B=1
+A.C=1 # This is a comment
+'''
+
+simple_config_with_invalid_entry_1 = '''
+A.A=FOO
+'''
+
+simple_config_with_invalid_entry_2 = '''
+A.B=2
+'''
+
+simple_config_with_invalid_entry_3 = '''
+A.C=
+'''
+
+simple_config_with_invalid_entry_4 = '''
+A.E==
+'''
+
+simple_config_with_invalid_entry_5 = '''
+A.D
+'''
+
+simple_config_with_invalid_entry_6 = '''
+A.D=1=
+'''
+
+
+class TestPICSChecker(unittest.TestCase):
+    @patch('builtins.open', mock_open(read_data=empty_config))
+    def test_empty_config(self):
+        pics_checker = PICSChecker('')
+        self.assertIsInstance(pics_checker, PICSChecker)
+
+    @patch('builtins.open', mock_open(read_data=simple_config))
+    def test_simple_config(self):
+        pics_checker = PICSChecker('')
+
+        self.assertFalse(pics_checker.check('A.A'))
+        self.assertFalse(pics_checker.check('A.DoesNotExist'))
+
+        self.assertTrue(pics_checker.check('A.B'))
+        self.assertTrue(pics_checker.check('A.b'))
+        self.assertTrue(pics_checker.check('a.b'))
+        self.assertTrue(pics_checker.check(' A.B'))
+        self.assertTrue(pics_checker.check('A.B '))
+
+    @patch('builtins.open', mock_open(read_data=simple_config))
+    def test_logic_negation(self):
+        pics_checker = PICSChecker('')
+        self.assertFalse(pics_checker.check('A.C'))
+        self.assertTrue(pics_checker.check('!A.C'))
+        self.assertFalse(pics_checker.check('!!A.C'))
+        self.assertTrue(pics_checker.check('!!!A.C'))
+
+    @patch('builtins.open', mock_open(read_data=simple_config))
+    def test_logical_and(self):
+        pics_checker = PICSChecker('')
+        self.assertFalse(pics_checker.check('A.C && A.B'))
+        self.assertFalse(pics_checker.check('A.C && A.B'))
+        self.assertTrue(pics_checker.check('!A.A && A.B'))
+
+    @patch('builtins.open', mock_open(read_data=simple_config))
+    def test_logical_or(self):
+        pics_checker = PICSChecker('')
+        self.assertFalse(pics_checker.check('A.A || A.C'))
+        self.assertTrue(pics_checker.check('A.B || A.C'))
+        self.assertTrue(pics_checker.check('!A.A || A.C'))
+        self.assertTrue(pics_checker.check('A.A || A.B || A.C'))
+
+    @patch('builtins.open', mock_open(read_data=simple_config))
+    def test_logical_parenthesis(self):
+        pics_checker = PICSChecker('')
+        self.assertFalse(pics_checker.check('(A.A)'))
+        self.assertTrue(pics_checker.check('(A.B)'))
+        self.assertTrue(pics_checker.check('!(A.A)'))
+        self.assertTrue(pics_checker.check('(!(A.A))'))
+        self.assertFalse(pics_checker.check('(A.A && A.B)'))
+        self.assertFalse(pics_checker.check('((A.A) && (A.B))'))
+        self.assertTrue(pics_checker.check('(!A.A && A.B)'))
+        self.assertTrue(pics_checker.check('(!(A.A) && (A.B))'))
+        self.assertTrue(pics_checker.check('(A.A || A.B)'))
```

Comment:
Might be good to have a test for something like `A && (B || C)` and the like too.